rootex/framework/entity_factory.cpp

```diff
@@ -313,3 +322,81 @@ void EntityFactory::deleteEntity(Ref<Entity> entity)
 	m_Entities.erase(entity->getID());
 	entity.reset();
 }
+
+String EntityFactory::saveEntityAsClass(Ref<Entity> entity)
+{
+	if (OS::IsDirectory("game/assets/classes/" + entity->getName()))
+	{
+		WARN("Class with name \"" + entity->getName() + "\" already exists. New Class not created.");
+		return nullptr;
+	}
+	else
+	{
+		OS::CreateDirectoryName("game/assets/classes/" + entity->getName());
+	}
+	return saveEntityAsClassRecursively(entity, "game/assets/classes/" + entity->getName() + "/");
+}
+
+String EntityFactory::saveEntityAsClassRecursively(Ref<Entity> entity, const String& path)
+{
+	Ref<HierarchyComponent> hierarchyComponent = entity->getComponent<HierarchyComponent>();
+	JSON::json entityJSON = entity->getJSON();
+	Vector<String> children;
+	for (EntityID child : hierarchyComponent->m_ChildrenIDs)
+	{
+		children.push_back(saveEntityAsClassRecursively(EntityFactory::GetSingleton()->getEntities().at(child), path));
+	}
+	entityJSON["Components"]["HierarchyComponent"]["children"] = children;
+	entityJSON["Components"]["HierarchyComponent"]["parent"] = ROOT_ENTITY_ID;
+	entityJSON["Entity"].erase("ID");
+	InputOutputFileStream file = OS::CreateFileName(path + entity->getName() + ".entity.json");
+	file << entityJSON.dump(4) << std::endl;
+	file.close();
+	return path + entity->getName() + ".entity.json";
+}
+
+Ref<Entity> EntityFactory::createEntityFromClass(const JSON::json& entityJSON)
+{
+	Ref<Entity> createdEntity = createEntityHierarchyFromClass(entityJSON);
+	fixParentID(createdEntity, ROOT_ENTITY_ID);
+	return createdEntity;
+}
+
+Ref<Entity> EntityFactory::createEntityHierarchyFromClass(JSON::json entityJSON)
+{
+	Vector<EntityID> ids;
+
+	for (String path : entityJSON["Components"]["HierarchyComponent"]["children"])
+	{
+		if (OS::IsFile(path))
+		{
+			JSON::json entityClass;
+			InputFileStream classFile(path);
+			classFile >> entityClass;
+			ids.push_back(createEntityHierarchyFromClass(entityClass)->getID());
+		}
+	}
+
+	entityJSON["Components"]["HierarchyComponent"]["children"] = ids;
+	Ref<Entity> newEntity = createEntity(entityJSON, "Error in creating" + entityJSON["Entity"]["Name"].dump());
+	if (newEntity)
+	{
+		return newEntity;
+	}
+	else
+	{
+		WARN("Could not create entity: " + entityJSON["Entity"]["Name"].dump());
+		return nullptr;
+	}
+}
+
+void EntityFactory::fixParentID(Ref<Entity> entity, EntityID id)
+{
+	Ref<HierarchyComponent> hierarchyComponent = entity->getComponent<HierarchyComponent>();
+	hierarchyComponent->m_ParentID = id;
+	setupEntity(entity);
```

Comment:
Entity::setupEntity() will return false in case it fails. Make sure to handle that case in EntityFactory::setupEntity()